detectable/src/main/java/com/blackduck/integration/detectable/detectables/gradle/inspection/parse/GradleReportLineParser.java

```diff
@@ -75,13 +75,23 @@ public GradleTreeNode parseLine(String line, Map<String, String> metadata) {
     }
 
     private String extractSubProjectName(String line) {
-        // A subProject dependency line looks exactly like: "+--- project :subProjectName"
-        if (line.contains(":")) {
-            String[] parts = line.split(":");
-            return parts[1].trim();
-        } else {
-            return "";
+        // A subProject dependency line looks exactly like: "+--- project :subProjectName" where subProjectName can
+        // be a nested subProject (for example  "+--- project :subProjectA:nestedSubProjectB:furtherNestedSubProjectC")
+        String[] parts = line.split(PROJECT_INDICATORS[0]);
+        if (parts.length == 2) {
+            // line looks as expected
+            String subprojName = parts[1].trim();
+            if (subprojName.startsWith(":")) {
+                // Drop the leading ":"
+                subprojName = subprojName.substring(1);
+                // In a Gradle dependencies tree, dependencies listed previously will have a " (*)" suffix
+                subprojName = removeSuffixes(subprojName);
+                return subprojName;
+            }
         }
+        // line didn't look as we expected
+        logger.debug("Could not extract subProject name from Gradle dependency tree report for line: " + line);
+        return "";
     }
```

Comment:
Refactored this method as per your suggestion, please re-review @andrian-sevastyanov